desktop/src/main/java/bisq/desktop/main/overlays/windows/ManualPayoutTxWindow.java

```diff
@@ -100,94 +198,549 @@ protected void setupKeyHandler(Scene scene) {
+        for (Dispute dispute :disputeObservableList) {
+            if (dispute.getDisputePayoutTxId() != null)    // only show disputes not paid out
+                continue;
+            if (!dispute.isClosed())    // only show closed disputes
+                continue;
+            if (recentTicketsOnly && dispute.getOpeningDate().toInstant().isBefore(twoWeeksAgo))
+                continue;
+            if (!disputeIds.contains(dispute.getTradeId()))
+                disputeIds.add(dispute.getTradeId());
+        }
+        disputeIds.sort((a, b) -> a.compareTo(b));
+        mediationDropDown.setItems(disputeIds);
+    }
+
+    private void clearInputFields() {
+        depositTxHex.setText("");
+        amountInMultisig.setText("");
         buyerPayoutAmount.setText("");
         sellerPayoutAmount.setText("");
-
         buyerAddressString.setText("");
-        buyerPubKeyAsHex.setText("");
-        buyerPrivateKeyAsHex.setText("");
-
         sellerAddressString.setText("");
+        buyerPubKeyAsHex.setText("");
         sellerPubKeyAsHex.setText("");
-        sellerPrivateKeyAsHex.setText("");
+    }
 
-        depositTxLegacy.setAllowIndeterminate(false);
-        depositTxLegacy.setSelected(false);
+    private boolean validateInputFields() {
+        return (depositTxHex.getText().length() == HEX_HASH_LENGTH &&
+                amountInMultisig.getText().length() > 0 &&
+                buyerPayoutAmount.getText().length() > 0 &&
+                sellerPayoutAmount.getText().length() > 0 &&
+                txFee.getText().length() > 0 &&
+                buyerAddressString.getText().length() > 0 &&
+                sellerAddressString.getText().length() > 0 &&
+                buyerPubKeyAsHex.getText().length() == HEX_PUBKEY_LENGTH &&
+                sellerPubKeyAsHex.getText().length() == HEX_PUBKEY_LENGTH);
+    }
 
-        actionButtonText("Sign and publish transaction");
+    private boolean validateInputFieldsAndSignatures() {
+        return (validateInputFields() &&
+                buyerSignatureAsHex.getText().length() > 0 &&
+                sellerSignatureAsHex.getText().length() > 0);
+    }
 
-        TxBroadcaster.Callback callback = new TxBroadcaster.Callback() {
-            @Override
-            public void onSuccess(@Nullable Transaction result) {
-                log.error("onSuccess");
-                UserThread.execute(() -> {
-                    String txId = result != null ? result.getTxId().toString() : "null";
-                    new Popup().information("Transaction successful published. Transaction ID: " + txId).show();
-                });
-            }
+    private void calculateTxFee() {
+        if (buyerPayoutAmount.getText().length() > 0 &&
+                sellerPayoutAmount.getText().length() > 0 &&
+                amountInMultisig.getText().length() > 0) {
+            Coin txFeeValue = Coin.parseCoin(amountInMultisig.getText())
```

Comment:
Coin.parseCoin can throw exceptions if input format is not as expected. Best to add a try catch...

Reply:
I'll wait for this before I cherry-pick.

Reply:
Resolved by https://github.com/bisq-network/bisq/pull/4899/commits/e888be319256ef1fd5a429a367274df35fed21b2